parser-java/kvs-archived-media-retrieval-sample/src/main/java/com/amazonaws/kinesisvideo/workers/ListFragmentWorker.java

```diff
@@ -0,0 +1,102 @@
+                              final FragmentSelector fragmentSelector) {
+        super(region, awsCredentialsProvider, streamName);
+        this.fragmentSelector = fragmentSelector;
+
+        amazonKinesisVideoArchivedMedia = AmazonKinesisVideoArchivedMediaClient
+                .builder()
+                .withCredentials(awsCredentialsProvider)
+                .withEndpointConfiguration(new AwsClientBuilder.EndpointConfiguration(endPoint, region.getName()))
+                .build();
+    }
+
+    public static ListFragmentWorker create(final String streamName,
+                                            final AWSCredentialsProvider awsCredentialsProvider,
+                                            final Regions region,
+                                            final AmazonKinesisVideo amazonKinesisVideo,
+                                            final FragmentSelector fragmentSelector) {
+        final GetDataEndpointRequest request = new GetDataEndpointRequest()
+                .withAPIName(APIName.LIST_FRAGMENTS).withStreamName(streamName);
+        final String endpoint = amazonKinesisVideo.getDataEndpoint(request).getDataEndpoint();
+
+        return new ListFragmentWorker(
+                streamName, awsCredentialsProvider, endpoint, region, fragmentSelector);
+    }
+
+    @Override
+    public List<String> call() {
+        try {
+            log.info("Start ListFragment worker on stream {}", streamName);
+
+            ListFragmentsRequest request = new ListFragmentsRequest()
+                    .withStreamName(streamName).withFragmentSelector(fragmentSelector).withMaxResults((long) 100);
+
+            ListFragmentsResult result = amazonKinesisVideoArchivedMedia.listFragments(request);
+
+
+            log.info("List Fragments called on stream {} response {} request ID {}",
+                    streamName,
+                    result.getSdkHttpMetadata().getHttpStatusCode(),
+                    result.getSdkResponseMetadata().getRequestId());
+
+            List<Fragment> fragments_list = result.getFragments();
+            String nextToken = result.getNextToken();
+
+            /* If result is truncated, keep making requests until nextToken is empty */
+            while (nextToken != null) {
+                request = new ListFragmentsRequest()
+                        .withStreamName(streamName).withNextToken(nextToken);
+                result = amazonKinesisVideoArchivedMedia.listFragments(request);
+
+                fragments_list = Stream.concat(fragments_list.stream(), result.getFragments().stream()).collect(Collectors.toList());
+                nextToken = result.getNextToken();
+            }
+
+            for (Fragment f: fragments_list) {
+                fragment_numbers.add(f.getFragmentNumber());
+            }
+
+            Collections.sort(fragment_numbers);
+
+        }
+        catch (Throwable t) {
+            System.out.println("ListFragmentWorker failed");
```

Comment:
I don't think this logging is necessary since you'll have another logging right after.